### ISSUE TITLE
Attempt extra credit

### ISSUE THREAD

**Issue by flip40** [CS149-3/homework3#20]
Customer leaves after waiting in queue for 10 seconds.

Likely implementation is to start a new thread (separate function) when customer arrives in queue. This thread simply does sleep(1) in a for loop for 10 cycles. If during any cycle it has been grabbed by a ticket seller (another function for ticket seller to call and another variable to track this), then the loop ends with nothing happening. If it ends the loop and still has not been grabbed by a ticketseller, then it can remove itself from the queue. This should be able to be done by something like `customer->ticketSeller->customerQueue.remove(this);`

This will need to make use of the queue mutex to be sure it has a lock on the queue when it removes itself
```
pthread_mutex_lock(&(customer->ticketSeller->queue_mutex));
customer->ticketSeller->customerQueue.remove(this);
pthread_mutex_unlock(&(customer->ticketSeller->queue_mutex));
```

**Comment by Seanv554** [comment]
Implemented in update_extracredit branch, ready to merge to master.

**Comment by flip40** [comment]
Seems to work great, has been merged to master. Thanks for doing the EC.
closed by #21
